velox/connectors/hive/HiveDataSource.cpp

```diff
@@ -782,7 +788,8 @@ std::shared_ptr<common::ScanSpec> HiveDataSource::makeScanSpec(
     // $bucket column. This filter is redundant and needs to be removed.
     // TODO Remove this check when Presto is fixed to not specify a filter
     // on $path and $bucket column.
-    if (auto name = pair.first.toString(); name == kPath || name == kBucket) {
+    if (auto name = pair.first.toString();
+        name == kPath || name == kBucket || metadataColumns.count(name) != 0) {
```

Comment:
Can you handle & remove such filters in Gluten?  This should be handled in split scheduling not inside table scan.  We will also fix it in presto_cpp at some point.

Reply:
@Yuhta sure, make sense.

Reply:
and apologize for so late to resolve the comment due to busy those week with internal project release work.